source/mesh_refinement/isosurfaces.cc

```diff
@@ -0,0 +1,391 @@
+                                values[index] = 1.0 - sum_composition;
+                            }
+                        }
+
+                      if ( isosurface.values_are_in_range(values))
+                        {
+                          // If the current refinement level is smaller or equal to the minimum
+                          // refinement level, any coarsening flags should be cleared.
+                          if (cell->level() <= isosurface.min_refinement)
+                            {
+                              clear_coarsen = true;
+                            }
+
+                          // If the current refinement level is smaller then the minimum
+                          // refinement level, a refinment flag should be placed.
+                          if (cell->level() <  isosurface.min_refinement)
+                            {
+                              refine = true;
+                              break;
+                            }
+
+                          // If the current refinement level is larger or equal to the maximum refinement
+                          // level, any refinement flag should be cleared.
+                          if (cell->level() >= isosurface.max_refinement)
+                            {
+                              clear_refine = true;
+                            }
+
+                          // If the current refinement level is larger then the maximum refinemment level,
+                          // a coarsening flag should be placed.
+                          if (cell->level() >  isosurface.max_refinement)
+                            {
+                              coarsen = true;
+                            }
+                        }
+                    }
+                }
+
+              // if both coarsen and refine are true, give preference to refinement
+              if (coarsen == true && refine == true)
+                coarsen = false;
+              if (refine == true)
+                clear_refine = false;
+              if (coarsen == true)
+                clear_coarsen = false;
+
+              // Perform the actual placement of the coarsening and refinement flags
+              // We want to make sure that the refinement never goes below the minimum
+              // or above the maximum, so we first check/set the coarsen/refine flag,
+              // and then check/set the clear coarsen/refine flag.
+              if (coarsen == true)
+                {
+                  cell->set_coarsen_flag ();
+                }
+              if (clear_coarsen == true)
+                {
+                  cell->clear_coarsen_flag ();
+                }
+              if (refine == true)
+                {
+                  cell->set_refine_flag ();
+                }
+              if (clear_refine == true)
+                {
+                  cell->clear_refine_flag ();
+                }
+            }
+          ++i;
```

Comment:
what is this used for? Maybe give it a more descriptive name?

Reply:
It wasn't used, so I removed it.